### PR TITLE
[CI/Build] vLLM cache directory for images

### PR DESCRIPTION
**Pull request by DarkLight1337**
This PR adds a cache directory for vLLM. Similar to HuggingFace, the location of the cache is given by the environment variable `VLLM_CACHE_ROOT`, or `$XDG_CACHE_HOME/vllm` if it is not provided (e.g. `~/.cache/vllm` by default). Under this directory, we have two subdirectories for containing different types of data:

- `VLLM_XLA_CACHE_PATH` for XLA persistent cache directory. cc @WoosukKwon 
- `VLLM_ASSETS_CACHE` for downloaded assets.
  - This removes the need to use `s3` or `curl` to download the images used in VLM examples and tests. cc @xwjiang2010 

Accordingly, `VLLM_CONFIG_ROOT` has been updated to take the `vllm` directory into account. This means that:
- The default has been changed from `$XDG_CONFIG_HOME` to `$XDG_CONFIG_HOME/vllm`.
- Subpaths of `VLLM_CONFIG_ROOT` should no longer prepend `vllm/` since that is included in the environment variable already.

### CODE REVIEW

**Comment by DarkLight1337** [comment]
@WoosukKwon do you think we can move the XLA cache under this directory as well?

File: vllm/assets/base.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+from pathlib import Path
+
+from vllm.envs import VLLM_ASSETS_CACHE
+
+
+def get_cache_dir():
+    """Get the path to the cache for storing downloaded assets."""
+    path = Path(VLLM_ASSETS_CACHE)
```
Comment (xwjiang2010) [inline vllm/assets/base.py:8]: need to expanduser here?

**Review by xwjiang2010 — APPROVED** [review]
just a small comment.

**Comment by WoosukKwon** [comment]
> do you think we can move the XLA cache under ~/.cache/vllm as well?

Yes. I think that's good for consistency.

**Comment by WoosukKwon** [comment]
BTW, we also have config-related cache under `~/.config/vllm/`.

File: vllm/assets/base.py (hunk 1 above)
Comment (DarkLight1337) [reply vllm/assets/base.py:8]: It is already done in `vllm.env` so no need to do it again here.

**Comment by DarkLight1337** [comment]
> BTW, we also have config-related cache under `~/.config/vllm/`.

The only cache I found under that directory is `gpu_p2p_access_cache_*`. I have moved it under `~/.cache/vllm`. cc @youkaichao

**Comment by youkaichao** [comment]
LGTM

**Review by WoosukKwon — APPROVED** [review]

**Pull request merged**